docs/guidelines-for-reviewing-datascience-code.md

```diff
@@ -1,25 +1,23 @@
+Code reviews are common practice in the software engineering field and have been for some time. They provide developers with a well-defined approach to sharing knowledge, ensuring exposure to new ideas and technologies, enforcing quality assurance of code and standards for the engineering process across both the codebase and the team itself. As members of the AI CoE in the office of the CTO, we believe that having a well-defined approach for how to do the same thing with data science and machine learning code is critical. Although, code is code, and we approach data science work as a subset of software engineering, there are some data science specific challenges associated with the code review process that should be addressed. Below are the 10 principles we use in our group to review data science work.
 
-1. **Understand the PR** : When asked for a review, make sure to first understand the issue linked and the goal of the pull request.
+1. **Understand the PR** : When asked for a review, make sure to first understand the issue linked and the goal of the pull request. Go through the Pull Request description to understand what has changed, a checklist of prerequisites like added documentation, notebooks, etc., and any additional notes the reviewer should be aware of.
 
 2. **NB Review** : Use tools like NB review when possible while reviewing Jupyter Notebooks, this will make reviewing the notebooks easier.
 
-3. **Don’t be afraid to ask questions** : When reviewing someone’s code, if you don't understand something or are curious about why a certain approach was implemented, ask questions and get clarification. 
+3. **Don’t be afraid to ask questions** : When reviewing someone’s code, if you don’t understand something or are curious about why a certain approach was implemented, ask questions, and get clarification.
 
 4. **Read Carefully** : Read the notebook thoroughly and try to look for logical errors or bugs in the code. Make sure you bring up any possible edge cases the code will not handle correctly.
 
-5. **Find ways to clean and condense the code** : Suggest alternate approaches to perform a similar action as the current code. This will present new techniques and various implementations for the team to consider.
+5. **Code Optimization** : Suggest alternate approaches to perform a similar action or ways to restructure nested functions/loops if possible. Instead of having one huge notebook, always look if it is possible to modularize the notebook into smaller notebooks. This will present new techniques and various implementations for the team to consider.
```

Comment:
This does not seem very specific to data science. I think this would be true about all code. Except the modular notebook bit, that said, Im not sure I agree that its _always_ best practice to divide a notebook. This section could be replaced.

Reply:
In my opinion, it gets hard for the reviewer to review a HUGE PR. So, maybe we could rephrase it a little or we could move this point to polishing the notebook, WDYT? @MichaelClifford

Reply:
It is difficult for a reviewer to review a huge PR. But dividing 1 big notebook into 3 does not actually make the PR any smaller. Also, I think it is the Data Scientist's responsibility to submit reasonably sized PR's. I still think this section can probably be replaced.

Reply:
Sure :+1: